pyiceberg/types.py

```diff
@@ -340,6 +340,12 @@ def __init__(
         data["doc"] = doc
         data["initial-default"] = data["initial-default"] if "initial-default" in data else initial_default
         data["write-default"] = data["write-default"] if "write-default" in data else write_default
+        if isinstance(data["type"], str):
+            try:
+                data["type"] = IcebergType.handle_primitive_type(data["type"], None)
+            except ValueError as e:
+                raise ValueError(f"Unsupported field type: {data['type']}.") from e
+
```

Comment:
nit: what do you think about using a pydantic field validator here? 
https://docs.pydantic.dev/latest/concepts/validators/#field-before-validator


```

    @validator("field_type", pre=True)
    def convert_field_type(cls, v):
        """Convert string values into IcebergType instances."""
        if isinstance(v, str):
            try:
                return IcebergType.handle_primitive_type(v, None)
            except ValueError as e:
                raise ValueError(f"Unsupported field type: {v}") from e
        return v  # If already an IcebergType, keep as is

```